proposals/lip-0050.md

```diff
@@ -171,14 +239,47 @@ registerBLSKeyParamsSchema = {
         "proofOfPossession": {
             "dataType": "bytes",
             "fieldNumber": 2
+        },
+        "generatorKey": {
+            "dataType": "bytes",
+            "fieldNumber": 3
         }
     }
 }
 ```
 
+#### Verification
+
+```python
+def verify(trs: Transaction) -> None:
+    validatorAddress = 20-byte address derived from trs.senderPublicKey
+
+    if validator.getValidatorAccount(validatorAddress) does not exists :
+        raise Exception('Public key does not correspond to a registered validator.')
+        
+    validatorAccount = validator.getValidatorAccount(validatorAddress)
+    if (validatorAccount.generatorKey != INVALID_ED25519_KEY
+        and validatorAccount.generatorKey == trs.params.generatorKey
+    ):
+        raise Exception('Input generator key does not equal the one set in the store.')
+
+    if validatorAccount.blsKey != INVALID_BLS_KEY:
```

Comment:
Can `blsKey` not exist?

Reply:
Yep, on Mainnet all BLS keys are set to the invalid one at regenesis.

Reply:
Not sure what you mean. As Jan said, if you don't have a BLS key, the regenesis process sets it to `INVALID`, but if you register after the migration, you must provide a valid BLS key.

Reply:
Can be resolved from my side.